### PR TITLE
#134/save search

### LINKED ISSUE #134: Implement saving current search
### Expected Behaviour

Now that we have a `SaveSearchButton.jsx` we need to implement it so that it saves a link with the current search query and any existing filters into the localstorage

### Current Behaviour

The `SaveSearchButton.jsx` is not implemented, although it is prepared to take the necessary handler functions

### Any other notes

_No response_

### PR DESCRIPTION
**Pull request by camelPhonso**
# Description

**Closes #134**  

The 'save search' button is now integrated in the `/search` page and updates state as we navigate through different search results.

### Files changed

- `App.jsx` now updates the searchParams when a search is made via the header
- `pages/Search.jsx` now includes the `SaveSearchButton.jsx` and declares the handler function for it that manages localstorage. Localstorage is also synched to a new state `isSaved`
- `SaveSearchButton.jsx` has been refactored to condition it's rendering on a state passed as a prop and to render the `.svg` directly instead of importing
- `Heart.svg` && `ReadHeart.svg` were removed as we are now rendering this directly in the component with the expected colour based on state

### UI changes

`/search` when not saved
Screenshot 2024-08-08 at 12 20 31

`/search` when saved
Screenshot 2024-08-08 at 12 21 02

### Changes to Documentation

None

# Tests

None

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import { useState, useEffect } from "react"
-import { Routes, Route, useNavigate } from "react-router-dom"
+import {
+  Routes,
+  Route,
+  useNavigate,
+  useSearchParams,
```
Comment (nichgalzin) [inline frontend/src/App.js:6]: This is unused in the app file and should be removed

File: frontend/src/App.js [hunk 2]
```diff
@@ -52,6 +57,7 @@ function App() {
             <Search
               searchResults={searchResults}
               searchStatus={searchStatus}
+              searchQuery={searchQuery}
```
Comment (nichgalzin) [inline frontend/src/App.js:60]: My IDE is saying the the navigate variable should be add to the useEffect dependency array

File: frontend/src/components/SaveSearchButton.jsx [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
-
-  const toggleHeart = () => {
-    if (isHeartClicked) {
-      onUnsave();
-    } else {
-      onSave();
-    }
-    setIsHeartClicked((prevIsHeartClicked) => !prevIsHeartClicked);
-  };
+import "../style/Heart.css"
```
Comment (nichgalzin) [inline frontend/src/components/SaveSearchButton.jsx:1]: There's a class in the Heart.css file called heart icon that has a transition on fill that might be nice to keep, although we can ditch the height and width from that class.

File: frontend/src/pages/Search.jsx [hunk 4]
```diff
@@ -1,7 +1,31 @@
...
+  searchResults,
+  searchStatus,
+  searchQuery,
+}) {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [isSaved, setIsSaved] = useState(false)
+
+  useEffect(() => {
+    searchQuery && setSearchParams({ query: searchQuery })
+  }, [])
```
Comment (nichgalzin) [inline frontend/src/pages/Search.jsx:18]: My IDE say this dependency array needs searchQuery and setSearchParams

**Review by nichgalzin — CHANGES_REQUESTED** [review]

File: frontend/src/App.js (hunk 2 above)
Comment (camelPhonso) [reply frontend/src/App.js:60]: done

File: frontend/src/App.js (hunk 1 above)
Comment (camelPhonso) [reply frontend/src/App.js:6]: done

File: frontend/src/pages/Search.jsx (hunk 4 above)
Comment (camelPhonso) [reply frontend/src/pages/Search.jsx]: done :)

File: frontend/src/components/SaveSearchButton.jsx (hunk 3 above)
Comment (camelPhonso) [reply frontend/src/components/SaveSearchButton.jsx:1]: done

**Review by nichgalzin — APPROVED** [review]

**Pull request merged**